src/test/run-pass/segfault-no-out-of-stack.rs

```diff
@@ -15,10 +15,59 @@ extern crate libc;
 use std::process::{Command, ExitStatus};
 use std::env;
 
+#[cfg(unix)]
+unsafe fn segfault() {
+    // Map and unmap a page of memory to guarantee that we have a pointer that
+    // will produce a segfault.
+    let pagesize = libc::sysconf(libc::_SC_PAGE_SIZE);
+    if pagesize == -1 {
+        panic!("Couldn't get pagesize");
+    }
+
+    let mut buf = libc::mmap(0 as *mut libc::c_void,
+                             pagesize as libc::size_t,
+                             libc::PROT_WRITE | libc::PROT_READ,
```

Comment:
Perhaps just do not pass in `PROT_WRITE` here to avoid the potent race condition where pointer could become valid between the unmap and dereference?

Not that it matters much.